src/profile-logic/marker-schema.js

```diff
@@ -394,8 +404,118 @@ export function formatFromMarkerSchema(
       return formatPercent(value);
     default:
       console.error(
-        `A marker schema of type "${markerType}" had an unknown format "${(format: empty)}"`
+        `A marker schema of type "${markerType}" had an unknown format ${JSON.stringify(
+          format
+        )}`
       );
       return value;
   }
 }
+
+export function isDOMRenderingFormat(format: MarkerFormatType): boolean {
+  return format === 'url' || typeof format === 'object' || format === 'list';
+}
+
+function _formatDOMFromMarkerSchema(
+  markerType: string,
+  format: MarkerFormatType,
+  value: any,
+  // add elements that allow interaction (like clickable urls)
+  supportsInteraction: boolean
+) {
```

Comment:
nit: can you type the return value, so that Flow can check whether we return the right value all the time? I believe it should be `React.Node`, and to use it I believe you need to `import * as React from 'react'` instead of `import React from 'react'` at the top. (I won't enter into details about why this works like this, this isn't very interesting...)